### PR TITLE
gnrc_netif_ethernet: fix debug output for received packet

### PR DESCRIPTION
**Pull request by miri64**
#12248 ## Contribution description
A received packet is outputted in DEBUG _after_ it was already parsed, but with a reference to the already parsed header. The result is that there can be some garbage in the output and the packet is not dumped in total. As without parsing we do not have access to the header yet, we
use the `gnrc_netif_addr_to_str()` helper function instead of parsing the destination address by hand.

### Testing procedure
Compile `examples/gnrc_networking` for `native` with `ENABLE_DEBUG=1` in `sys/net/gnrc/netif/ethernet/gnrc_netif_ethernet.c`. Send some packets to the application e.g. using `ping6`. You will get something like

```
gnrc_netif_ethernet: received packet from 02:ca:6d:11:65:c4 of length 78
00000000  02  58  97  2A  A6  2D  02  CA  6D  11  65  C4  86  DD  60  00
00000010  00  00  00  18  3A  FF  FE  80  00  00  00  00  00  00  00  CA
00000020  6D  FF  FE  11  65  C4  FE  80  00  00  00  00  00  00  00  58
00000030  97  FF  FE  2A  A6  2D  88  00  5A  3A  40  00  00  00  FE  80
00000040  00  00  00  00  00  00  00  CA  6D  FF  FE  11  65  C4
```

On master however, you get something like

```
gnrc_netif_ethernet: received packet from 02:ca:6d:11:65:c4 of length 78
00000000  02  58  97  2A  A6  2D  02  CA  6D  11  65  C4  86  DD  65  C4
00000010  60  00  00  00  00  18  3A  FF  FE  80  00  00  00  00  00  00
00000020  00  CA  6D  FF  FE  11  65  C4  FE  80  00  00  00  00  00  00
00000030  00  58  97  FF  FE  2A  A6  2D  88  00  5A  3A  40  00  00  00
00000040  FE  80  00  00  00  00  00  00  00  CA  6D  FF  FE  11
```

with `65  C4` in the first line of the dump being padding added by the header parsing function and the packet ending early (incidentally `65  C4` is missing at the end in my example).

(note: the dumps are from an `samr21-xpro` running `ethos`, so your dumps might vary)

### Issues/PRs references
Found while debugging for https://github.com/RIOT-OS/RIOT/pull/11596#issuecomment-531739735, but unrelated.

### CODE REVIEW

**Comment by miri64** [comment]
Running tests, because I'm not sure if any are using this output and `gnrc_netif_addr_to_str()` has a different case than the previous code.

File: sys/net/gnrc/netif/ethernet/gnrc_netif_ethernet.c [hunk 1]
```diff
@@ -34,6 +34,8 @@
 static int _send(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt);
 static gnrc_pktsnip_t *_recv(gnrc_netif_t *netif);
 
+static char addr_str[ETHERNET_ADDR_LEN * 3];
```
Comment (benpicco) [inline sys/net/gnrc/netif/ethernet/gnrc_netif_ethernet.c]: unused if `ENABLE_DEBUG` is not set.

File: sys/net/gnrc/netif/ethernet/gnrc_netif_ethernet.c [hunk 2]
```diff
@@ -196,6 +198,12 @@ static gnrc_pktsnip_t *_recv(gnrc_netif_t *netif)
             gnrc_pktbuf_realloc_data(pkt, nread);
         }
```
Comment (benpicco) [inline sys/net/gnrc/netif/ethernet/gnrc_netif_ethernet.c]: Why not put `addr_str[]` here?

Comment (miri64) [reply sys/net/gnrc/netif/ethernet/gnrc_netif_ethernet.c]: A larger array like this I rather put into global space as it is done through-out the rest of the code base.

File: sys/net/gnrc/netif/ethernet/gnrc_netif_ethernet.c (hunk 1 above)
Comment (benpicco) [reply sys/net/gnrc/netif/ethernet/gnrc_netif_ethernet.c]: I thought it would generate a warning about the unused variable, but I did not test this.

Comment (benpicco) [reply sys/net/gnrc/netif/ethernet/gnrc_netif_ethernet.c]: Elsewhere in the code `IPV6_ADDR_MAX_STR_LEN` is used instead.

**Pull request merged**